File: Data Structures & Algorithms/max-water-container/submission-2.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let left = 0 , right = heights.length -1 ;
        let  ans  = 0 ;
        while(left<right){
            let  area  = Math.min(heights[left],heights[right])* (right -left);
            ans = Math.max(ans,area);
            if(heights[left]<heights[right])left++;
            else if(heights[right]<heights[left])right--;
            else {
                left++;
                right--;
            }

        }
        return ans  ;
    }
}

File: Data Structures & Algorithms/jump-game/submission-2.java
class Solution {
    public boolean canJump(int[] nums) {
         int idx  = 0 ;
    for(int i = 0 ; i< nums.length ; i++){
        if(idx < i) return false ;
           idx = Math.max(idx , i + nums[i]);
     }
         return (idx >= nums.length -1) ;
    }
}
